docker/prepare_proxy.sh

```diff
@@ -79,12 +108,15 @@ iptables -t nat -A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN                     -m
 # All outbound traffic will be redirected to Envoy by default. If
 # IP_RANGES_INCLUDE is non-empty, only traffic bound for the
 # destinations specified in this list will be captured.
-IFS=,
+# IF IP_RANGES_EXCLUDE is non-empty, traffic specified by it will
+# NOT be redirected to the sidecar.
+# NOTE: IP_RANGES_INCLUDE and IP_RANGES_EXCLUDE cannot be specified.
 if [ "${IP_RANGES_INCLUDE}" != "" ]; then
-    for cidr in ${IP_RANGES_INCLUDE}; do
-        iptables -t nat -A ISTIO_OUTPUT -d ${cidr} -j ISTIO_REDIRECT          -m comment --comment "istio/redirect-ip-range-${cidr}"
-    done
+    output_redir ${IP_RANGES_INCLUDE} "include"
     iptables -t nat -A ISTIO_OUTPUT -j RETURN                                 -m comment --comment "istio/bypass-default-outbound"
+elif [ "${IP_RANGES_EXCLUDE}" != "" ]; then
+    output_redir ${IP_RANGES_EXCLUDE} "exclude"
+    iptables -t nat -A ISTIO_OUTPUT -j ISTIO_REDIRECT                         -m comment --comment "istio/bypass-exclude-default-outbound"
```

Comment:
Duplicating the `for` loop and removing the extra function (w/conditional) seems cleaner and results in less code, e.g.

```bash
if [ "${IP_RANGES_INCLUDE}" != "" ]; then
    for cidr in ${IP_RANGES_INCLUDE}; do
        iptables -t nat -A ISTIO_OUTPUT -d ${cidr} -j ISTIO_REDIREC
    done     
elif [ "${IP_RANGES_EXCLUDE}" != "" ]; then
    for cidr in ${IP_RANGES_EXCLUDE}; do
        iptables -t nat -A ISTIO_OUTPUT -d ${cidr} -j RETURN
    done
    iptables -t nat -A ISTIO_OUTPUT -j ISTIO_REDIRECT  
else
    iptables -t nat -A ISTIO_OUTPUT -j ISTIO_REDIRECT
fi

```